components/list/list-item-generic-layout.js

```diff
@@ -39,23 +49,41 @@ class ListItemGenericLayout extends RtlMixin(LitElement) {
 					[start outside-control-start] minmax(0, min-content)
 					[control-start outside-control-end] minmax(0, min-content)
 					[control-end content-start] auto
-					[content-end actions-start] auto
+					[content-end actions-start] minmax(0, max-content)
 					[end actions-end];
 			}
+			:host([front-row]) ::slotted([slot="outside-control"]),
+			:host([front-row]) ::slotted([slot="control"]),
+			:host([front-row]) ::slotted([slot="content"]) {
+				z-index: 2;
+			}
+			:host([front-row]) ::slotted([slot="outside-control-action"]) {
+				z-index: 3;
+			}
+			:host([front-row]) ::slotted([slot="control-action"]) {
+				z-index: 4;
+			}
+			:host([front-row]) ::slotted([slot="content-action"]) {
+				z-index: 5;
+			}
+			:host([front-row]) ::slotted([slot="actions"]) {
+				z-index: 6;
+			}
+
```

Comment:
It's possible I'm missing something and there's an easier way to do this by targeting the host, but I couldn't find one.

Reply:
Yeah it's not great to sprout attributes for internals on the host, given that the host attributes are sorta part of the public API... subject to be manipulated by consumers if used incorrectly.  That being said, we'd need to apply this attribute or a class to an internal element to avoid it in this case... and in this case, we only have slots, and I'm not 100% sure we can apply custom attributes to slots and effectively target them with CSS in all browsers.

Reply:
After looking into alternative options, the best thing to do was to reflect an obviously "private" attribute on the host.